### DISCUSSION TITLE
React-native support for the rules engine

### DISCUSSION THREAD

**Discussion by prasunsrivastava** [gorules/zen#187] [Q&A]
I am looking to integrate the zen engine on frontend to run the rules directly from the mobile app. I understand that this is not possible as of today as the js package is specifically for the backend node.js codebase. I tried converting the npm package to run on the frontend by using Browserify. However, the conversion fails while loading the native bindings. Can someone in the community point me in the right direction to convert the native rust code to a frontend JS binding? Any help would be appreciated. Thanks.

**Comment by ivanmiletic** [comment]
Hi, this is not yet possible or at least we didn't try building packages for mobile architecture, we currently support only Python, Rust, Go and Node on a few architectures/platforms.

We do have plans to expand to WASM, iOS and Android but it is not in plan for 2024, at least we didn't get request from our enterprise customers or community.

Next languages that will be supported will be dotNet core and Java.

**Reply by prasunsrivastava** [reply]
Thanks for the quick response. However, would it be possible to point me in the right direction if I were to make a react-native support build from the core files? I am willing to put in the effort if it can be done in 50-60 hours. It might help others in the community too.

**Reply by ivanmiletic** [reply]
We didn't have time yet to investigate but you will need rust tools for compiling for iOS and Android, also exposing our native API to iOS/Android api:

Please note this is something that i just googled:
Example rust in iOS: https://krirogn.dev/blog/integrate-rust-in-ios

Next, you will need to expose zen-engine api to iOS, you can see how we do it for C on this link: https://github.com/gorules/zen/tree/master/bindings/c

Even with this we are not sure that all of our underlying libraries (e.g. bumpalo, quickjs etc.) would support iOS/Android at this stage.

**Reply by prasunsrivastava** [reply]
Thanks a lot for these links. Will take a look and keep you posted.

**Comment by ivanmiletic** [comment]
No plans to directly support React Native however we will be releasing Kotlin (Android) and Swift (iOS) support by EOY

**Reply by ivanmiletic** [reply]
Hi @prasunsrivastava , we have officially added support for iOS Swift and Android Kotlin, we have also added WASM support. We do not have strong use case for directly supporting React Native at this stage, however you can build your bridge using our official libraries:

gorules/zen-ios (git)
zen-engine-kotlin-android from Maven for Kotlin bindings
